extensions/wikia/UserRenameTool/UserRenameToolProcess.class.php

```diff
@@ -246,197 +194,358 @@ protected function renameAccount() {
+
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * @param string $oldUserName
+	 * @param User $oldUser
+	 *
+	 * @return int
+	 */
+	private function getOldUserId( $oldUserName, $oldUser ) {
+		global $wgContLang, $wgCapitalLinks;
 
 		// Check for the existence of lowercase oldusername in database.
 		// Until r19631 it was possible to rename a user to a name with first character as lowercase
-		if ( $oldTitle->getText() !== $wgContLang->ucfirst( $oldTitle->getText() ) ) {
+		if ( $oldUserName !== $wgContLang->ucfirst( $oldUserName ) ) {
 			// oldusername was entered as lowercase -> check for existence in table 'user'
 			$dbr = WikiFactory::db( DB_SLAVE );
-			$uid = $dbr->selectField( '`user`', 'user_id',
-				array( 'user_name' => $oldTitle->getText() ),
-				__METHOD__ );
+			$uid = $dbr->selectField(
+				'`user`',
+				'user_id',
+				[ 'user_name' => $oldUserName ],
+				__METHOD__
+			);
 
-			$this->addLog( 'Running query: ' . $dbr->lastQuery() . " resulted in " . $dbr->affectedRows() . " row(s) being affected." );
+			$this->addLog(
+				"Running query: " .
+				$dbr->lastQuery() .
+				" resulted in " .
+				$dbr->affectedRows() .
+				" row(s) being affected."
+			);
 
 			if ( $uid === false ) {
 				if ( !$wgCapitalLinks ) {
-					$uid = 0; // We are on a lowercase wiki but lowercase username does not exists
+					// We are on a lowercase wiki, but lowercase username does not exists
+					return 0;
 				} else {
 					// We are on a standard uppercase wiki, use normal
-					$uid = $oldUser->idForName();
-					$oldTitle = Title::makeTitleSafe( NS_USER, $oldUser->getName() );
+					return $oldUser->idForName();
 				}
 			}
-		} else {
-			// oldusername was entered as upperase -> standard procedure
-			$uid = $oldUser->idForName();
+
+			return $uid;
 		}
 
-		$this->addInternalLog( "id: uid={$uid} old={$oldUser->getName()}:{$oldUser->getId()} new={$newUser->getName()}:{$newUser->getId()}" );
+		// oldusername was entered as upperase -> standard procedure
+		return $oldUser->idForName();
+	}
 
+	/**
+	 * @param int $uid
+	 * @param User $newUser
+	 * @param User $oldUser
+	 * @param Title $oldTitle
+	 *
+	 * @return bool
+	 */
+	private function doStuffIfNewUserNameAlreadyExists( $uid, $newUser, $oldUser, $oldTitle ) {
+		$repeating = false;
+
+		// invalidate properties cache and reload to get updated data
+		// needed here, if the cache is wrong bad things happen
+		$this->addInternalLog( "pre-invalidate: titletext={$oldTitle->getText()} old={$oldUser->getName()}" );
+
+		$oldUser->invalidateCache();
+		$oldUser = User::newFromName( $oldTitle->getText(), false );
+
+		$renameData = $oldUser->getGlobalAttribute( 'renameData', '' );
+
+		$this->addInternalLog(
+			"post-invalidate: titletext={$oldTitle->getText()} old={$oldUser->getName()}:{$oldUser->getId()}"
+		);
+
+		$this->addLog( "Scanning user option renameData for process data: {$renameData}" );
+
+		if ( stripos( $renameData, self::RENAME_TAG ) !== false ) {
```

Comment:
I think it would be more clear to use a regular expression here, e.g.:
```php
if ( preg_match_all("/([^=]+)=([^;]+);?/", $renameData, $out, PREG_SET_ORDER) )  {
    foreach ( $out as $tuple ) {
        list( $full, $tagName, $value ) = $tuple;

        if ( $tagName == self::RENAME_TAG ) {
            $repeating = $value == $newUser->getName() 
        }

        if ( $tagName == self:: PHALANX_BLOCK_TAG ) {
            $this->mPhalanxBlockId = (int) $value;
        }
    } 
}
```